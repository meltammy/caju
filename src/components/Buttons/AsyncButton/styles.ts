import styled from "styled-components";
import { Spinner } from "~/components/Spinner";

export const StyledSpinner = styled(Spinner)`
  width: 5px;
  height: 5px;
  border-width: 3px;
`;
