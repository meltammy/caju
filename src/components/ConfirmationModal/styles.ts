import styled from "styled-components";

export const ModalOverlay = styled.div<{ $isOpen: boolean }>`
  display: ${(props) => (props.$isOpen ? "flex" : "none")};
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  justify-content: center;
  align-items: center;
`;

export const ModalContainer = styled.div`
  background: white;
  border-radius: 8px;
  padding: 20px;
  min-width: 500px;
  max-width: 95vw;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  z-index: 999;

  @media screen and (max-width: 1200px) {
    min-width: 50vw;
    margin: 1rem;
  }
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 20px;
`;

export const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;

  @media screen and (max-width: 1200px) {
    flex-direction: column-reverse;
  }

  button {
    width: 100%;
    place-content: center;
  }
`;
