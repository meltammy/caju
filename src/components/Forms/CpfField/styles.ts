import ReactInputMask from "react-text-mask";
import styled from "styled-components";

export const Input = styled(ReactInputMask)`
  width: 100%;
  padding: 0 0.5rem;
  min-height: 2.25rem;
  background-color: ${({ theme }) => theme.colors.white};
  transition: all 0.2s ease-in-out 0s;

  vertical-align: middle;
  border: 1px solid rgba(36, 28, 21, 0.3);
  border-radius: 0.5rem;

  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: normal;

  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.petroleumBlue};
    box-shadow: inset 0 0 0 1px ${({ theme }) => theme.colors.petroleumBlue};
  }
`;
