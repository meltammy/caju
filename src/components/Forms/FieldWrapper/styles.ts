import styled from "styled-components";

export const ErrorMessage = styled.span`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 0.75rem;
`;

export const Container = styled.div`
  width: 100%;
`;

export const Label = styled.label`
  width: 100%;
  font-weight: 600;
`;
