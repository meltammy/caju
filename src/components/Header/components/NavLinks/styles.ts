import styled from "styled-components";

export const NavItems = styled.a`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
