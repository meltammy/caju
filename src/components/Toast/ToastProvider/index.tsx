import { ReactNode, useState } from "react";
import { Toast } from "../types";
import ToastContext from "../ToastContext";
import ToastContainer from "..";

type Props = {
  children: ReactNode;
};

export function ToastProvider({ children }: Props) {
  const [toasts, setToasts] = useState<Toast[]>([]);
  const [nextId, setNextId] = useState(0);

  const removeToast = (id: number) => {
    setToasts((prev) => prev.filter((toast) => toast.id !== id));
  };

  const addToast = (message: string, type: "success" | "error" | "info") => {
    const id = nextId;
    setToasts((prev) => [...prev, { id, message, type }]);
    setNextId((prev) => prev + 1);

    setTimeout(() => {
      removeToast(id);
    }, 5000);
  };

  return (
    <ToastContext.Provider value={{ addToast }}>
      <ToastContainer toasts={toasts} onRemove={removeToast} />
      {children}
    </ToastContext.Provider>
  );
}
