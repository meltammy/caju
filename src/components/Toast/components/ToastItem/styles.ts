import styled from "styled-components";
import { ToastType } from "../../types";

export const ToastStyled = styled.div<{ type: ToastType }>`
  background-color: ${({ theme, type }) => theme.colors[type]};
  color: white;
  padding: 10px;
  margin: 10px 0;
  border-radius: 5px;
  transition: opacity 0.5s;

  button {
    margin-left: 10px;
    background: transparent;
    border: none;
    color: white;
    cursor: pointer;
  }
`;
