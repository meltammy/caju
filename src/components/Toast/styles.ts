import styled from "styled-components";

export const Container = styled.div`
  position: fixed;
  bottom: 0;
  left: 2rem;
  z-index: 99;
`;
