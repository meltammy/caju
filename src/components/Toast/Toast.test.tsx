import "@testing-library/jest-dom";

import { render, fireEvent, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "~/../styles/theme";
import { act } from "react";
import { useToast } from "./ToastContext";
import { ToastProvider } from "./ToastProvider";

const message = "Test message";
const Button = () => {
  const { addToast } = useToast();

  return (
    <button onClick={() => addToast(message, "success")}>Show Toast</button>
  );
};

const renderComponent = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ToastProvider>
        <Button />
      </ToastProvider>
    </ThemeProvider>
  );
};

describe("ToastProvider component", () => {
  test("renders toast messages", () => {
    renderComponent();

    const button = screen.getByText("Show Toast");
    fireEvent.click(button);

    const toastMessage = screen.getByText(message);
    expect(toastMessage).toBeInTheDocument();
  });

  test("removes toast after timeout", async () => {
    jest.useFakeTimers();

    renderComponent();

    const button = screen.getByText("Show Toast");
    fireEvent.click(button);

    const toastMessage = screen.getByText(message);
    expect(toastMessage).toBeInTheDocument();

    act(() => jest.runAllTimers());

    expect(screen.queryByText(message)).not.toBeInTheDocument();
  });

  test("closes toast when button is clicked", () => {
    renderComponent();

    const button = screen.getByText("Show Toast");
    fireEvent.click(button);

    const closeButton = screen.getByRole("button", { name: /x/i });
    fireEvent.click(closeButton);

    expect(screen.queryByText(message)).not.toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = renderComponent();

    const button = screen.getByText("Show Toast");
    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
  });
});
