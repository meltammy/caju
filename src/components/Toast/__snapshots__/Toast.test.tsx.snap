// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastProvider component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb iwaunG"
  >
    <div
      class="sc-gtLWhw jtmJAR"
      id="0-toast"
      type="success"
    >
      Test message
      <button>
        X
      </button>
    </div>
  </div>
  <button>
    Show Toast
  </button>
</DocumentFragment>
`;
